backend/tests/unit/api/Controllers/Research/ResearchControllerTest.cs

```diff
@@ -0,0 +1,165 @@
+
+namespace Pims.Api.Test.Controllers.Research
+{
+    [Trait("category", "unit")]
+    [Trait("category", "api")]
+    [Trait("group", "research")]
+    [ExcludeFromCodeCoverage]
+    public class SearchControllerTest
+    {
+        #region Variables
+        public readonly static IEnumerable<object[]> ResearchFilters = new List<object[]>()
+        {
+            new object [] { new SModel.ResearchFilterModel() { RegionCode = 1 } },
+            new object [] { new SModel.ResearchFilterModel() { RFileNumber = "1234" } },
+            new object [] { new SModel.ResearchFilterModel() { Name = "rname" } },
+        };
+
+        public readonly static IEnumerable<object[]> ResearchQueryFilters = new List<object[]>()
+        {
+            new object [] { new Uri("http://host/api/research/search?RegionCode=1") },
+            new object [] { new Uri("http://host/api/research/search?RFileNumber=100-000-000") },
+            new object [] { new Uri("http://host/api/research/search?Name=rname") },
+        };
+        #endregion
+
+        #region Tests
+        #region GetResearchFiles
+        /// <summary>
+        /// Make a successful request that includes the research filter.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(ResearchFilters))]
+        public void GetResearchFiles_All_Success(SModel.ResearchFilterModel filter)
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var controller = helper.CreateController<SearchController>(Permissions.LeaseView);
+
+            var researchFiles = new[] { EntityHelper.CreateResearchFile(1) };
+
+            var service = helper.GetService<Mock<IPimsService>>();
+            var mapper = helper.GetService<IMapper>();
+
+            service.Setup(m => m.ResearchFileService.GetPage(It.IsAny<ResearchFilter>())).Returns(new Paged<Entity.PimsResearchFile>(researchFiles));
+
+            // Act
+            var result = controller.GetResearchFiles(filter);
+
+            // Assert
+            var actionResult = Assert.IsType<JsonResult>(result);
+            var actualResult = Assert.IsType<Models.PageModel<ResearchFileModel>>(actionResult.Value);
+            var expectedResult = mapper.Map<Models.PageModel<ResearchFileModel>>(new Paged<Entity.PimsResearchFile>(researchFiles));
+            Assert.Equal(expectedResult, actualResult, new DeepPropertyCompare());
+            service.Verify(m => m.ResearchFileService.GetPage(It.IsAny<ResearchFilter>()), Times.Once());
+        }
+
+        /// <summary>
+        /// Make a successful request that passes the filter in the query string.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(ResearchQueryFilters))]
+        public void GetResearchFiles_Query_Success(Uri uri)
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var controller = helper.CreateController<SearchController>(Permissions.PropertyView, uri);
```

Comment:
Not sure how this test passes as the controller requires LeaseView permission instead?

Reply:
thanks for the reminder, need to update these now that I have merged with Manuel's code.